kyo-zio/shared/src/main/scala/kyo/zios.scala

```diff
@@ -0,0 +1,68 @@
+package kyo
+
+import ZIOs.internal.*
+import core.*
+import core.internal.*
+import kyo.fibersInternal.*
+import scala.util.control.NonFatal
+import zio.Task
+import zio.ZIO
+
+opaque type ZIOs <: Fibers = Tasks & Fibers
+
+object ZIOs:
+
+    def get[E >: Nothing: Tag, A](v: ZIO[Any, E, A]): A < (Aborts[E] & ZIOs) =
+        val task = v.fold[A < Aborts[E]](e => Aborts.fail(e), a => a)
+        Tasks.suspend[A < Aborts[E], A, Aborts[E]](task, identity)
+
+    def get[A](v: ZIO[Any, Nothing, A]): A < ZIOs =
+        Tasks.suspend[A](v)
+
+    inline def get[R: zio.Tag, A](v: ZIO[R, ?, A])(using tag: Tag[Envs[R]]): A < (Envs[R] & ZIOs) =
+        compiletime.error("ZIO environments are not supported yet. Please handle them before calling this method.")
+
+    def run[T: Flat](v: T < (Aborts[Throwable] & ZIOs)): Task[T] =
+        def loop[U](v: U < ZIOs): Task[U] =
+            v match
+                case kyo: Suspend[?, ?, ?, ?] =>
+                    try
+                        if kyo.tag == Tag[IOs] then
+                            val k = kyo.asInstanceOf[Suspend[?, Unit, U, Fibers]]
+                            ZIO.suspend(loop(k(())))
+                        else if kyo.tag == Tag[FiberGets] then
+                            val k = kyo.asInstanceOf[Suspend[Fiber, Any, U, FiberGets]]
+                            k.command match
+                                case Done(v) =>
+                                    ZIO.suspend(loop(k(v)))
+                                case Promise(p) =>
+                                    ZIO.asyncInterrupt[Any, Throwable, U] { cb =>
+                                        p.onComplete(v => cb(ZIO.suspend(loop(v.map(k)))))
+                                        Left(ZIO.succeed(p.interrupt()))
+                                    }
+                            end match
+                        else if kyo.tag == Tag[Tasks] then
+                            val k = kyo.asInstanceOf[Suspend[Task, Any, U, ZIOs]]
+                            k.command.flatMap(v => loop(k(v)))
+                        else
+                            bug.failTag(kyo.tag, Tag[Fibers & ZIOs])
+                    catch
+                        case ex if NonFatal(ex) =>
+                            ZIO.fail(ex)
+                case v =>
+                    ZIO.succeed(v.asInstanceOf[U])
+            end match
+        end loop
+
+        loop(Aborts.run(v)).flatMap {
+            case Left(ex)     => ZIO.fail(ex)
+            case Right(value) => ZIO.succeed(value)
+        }
```

Comment:
I believe this is correct, but I want to ensure that Aborts short circuit in the correct order. I was trying to add a test myself, but was having issues with the scalatest runner.

```
        val a = Aborts.fail(new RuntimeException("Aborts"))
        val b = ZIOs.get(ZIO.fail(new RuntimeException("ZIOs")))

        def run(expected: String, v: Nothing < (Aborts[Throwable] & ZIOs)): Task[Assertion] =
            ZIOs.run(v).fold(
                e => assert(e.getMessage == expected),
                _ => fail("Unexpected success")
            )

        run("Aborts", a.map(_ => b)) *> run("ZIOs", a.map(_ => b))
```

It's important to have this test to ensure future refactorings don't break compatibility

Reply:
I've added a couple tests for this.